README.md

```diff
@@ -370,6 +374,42 @@ end
 
 It is particularly useful when writing tests.
 
+
+Design Details
+--------------
+
+#### `Mercury#republish`
+
+This method publishes a copy of the message to the _back_ of the
+queue, then acks the original message. This is a similar operation to
+ack/nack/reject. It is only applicable to messages received by
+workers, since listener messages cannot be acknowledged. Unlike other
+acknowledgements, `#republish` takes a continuation (due to the
+publish operation), so the method is located on `Mercury` rather than
+`Mercury::ReceivedMessage`.
```

Comment:
So can this not be synchronous as in `Mercury::Sync`? I suppose the only reason to have it be synchronous is for logging purposes. If there is a failure on the republish, I wanted to log CRITICAL. I was able to count on that for the synchronous call.

If this asynchronous call is made, will it raise an error if the rabbit server is down or the connection is broken? If so, then I can still log my CRITICAL error upon the failed republish.